lab_5_scrapper/scrapper.py

```diff
@@ -90,75 +214,151 @@ def __init__(self, config: Config) -> None:
         """
         Initializes an instance of the Crawler class
         """
-        pass
+        self.config = config
+        self.urls = []
 
     def _extract_url(self, article_bs: BeautifulSoup) -> str:
         """
         Finds and retrieves URL from HTML
         """
-        pass
+        links = article_bs.find_all('a')
+        for link in links:
+            url = link.get('href')
+            if url and url.count('/') == 4 and url[:9] == '/novosti/':
+                url = 'https://www.zebra-tv.ru' + url
+                if url not in self.urls and len(self.urls) < self.config.get_num_articles():
+                    self.urls.append(url)
```

Comment:
extract url should not write to internal state, it should return a single url for each article block in the list of articles on the seed page